### PR TITLE
Bumps version to 3.4

### PR DESCRIPTION
**Pull request by ibesora**
This also updates the changelog to add info about the custom raster-dem encodings

### CODE REVIEW

**Review by neodescis — APPROVED** [review]
Looks good to me. @HarelM , are we ready to bump version and do a release?

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,23 @@
 - 3d buildings example doesn't work as expected ([#3165](https://github.com/maplibre/maplibre-gl-js/pull/3165))
 - _...Add new stuff here..._
 
+## 3.3.2
+
+### ✨ Features and improvements
```
Comment (HarelM) [inline CHANGELOG.md:21]: This is duplicated... no need to duplicate, just add empty sections under "main" instead.

**Comment by HarelM** [comment]
Sure, if someone wants this version out I see no reason to keep it.
We can always release another one soon after.

File: CHANGELOG.md (hunk 1 above)
Comment (ibesora) [reply CHANGELOG.md:13]: I thought I had removed them 🤦

**Comment by ibesora** [comment]
Wondering if this should be a new minor version instead of a patch one. 
There are things in the changelog that seem to be new features under patch versions so I'm not sure (i.e. adding `cache` to `RequestParameters`in 3.2.2) but happy to follow more strict semver rules if needed

**Comment by HarelM** [comment]
Yes, let's do 3.4.

**Comment by ibesora** [comment]
> Yes, let's do 3.4.

Done!

**Review by HarelM — APPROVED** [review]

**Pull request merged**